lib/model/country.dart

```diff
@@ -1,368 +1,17 @@
-      case bi:
-        return context.strings.country_bi;
-      case bj:
-        return context.strings.country_bj;
-      case bl:
-        return context.strings.country_bl;
-      case bm:
-        return context.strings.country_bm;
-      case bn:
-        return context.strings.country_bn;
-      case bo:
-        return context.strings.country_bo;
-      case bq:
-        return context.strings.country_bq;
-      case br:
-        return context.strings.country_br;
-      case bs:
-        return context.strings.country_bs;
-      case bt:
-        return context.strings.country_bt;
-      case bv:
-        return context.strings.country_bv;
-      case bw:
-        return context.strings.country_bw;
-      case by:
-        return context.strings.country_by;
-      case bz:
-        return context.strings.country_bz;
-      case ca:
-        return context.strings.country_ca;
-      case cc:
-        return context.strings.country_cc;
-      case cd:
-        return context.strings.country_cd;
-      case cf:
-        return context.strings.country_cf;
-      case cg:
-        return context.strings.country_cg;
-      case ch:
-        return context.strings.country_ch;
-      case ci:
-        return context.strings.country_ci;
-      case ck:
-        return context.strings.country_ck;
-      case cl:
-        return context.strings.country_cl;
-      case cm:
-        return context.strings.country_cm;
-      case cn:
-        return context.strings.country_cn;
-      case co:
-        return context.strings.country_co;
-      case cr:
-        return context.strings.country_cr;
-      case cu:
-        return context.strings.country_cu;
-      case cv:
-        return context.strings.country_cv;
-      case cw:
-        return context.strings.country_cw;
-      case cx:
-        return context.strings.country_cx;
-      case cy:
-        return context.strings.country_cy;
-      case cz:
-        return context.strings.country_cz;
-      case de:
-        return context.strings.country_de;
-      case dj:
-        return context.strings.country_dj;
-      case dk:
-        return context.strings.country_dk;
-      case dm:
-        return context.strings.country_dm;
-      // case dom:
-      //return context.strings.country_do;
-      case dz:
-        return context.strings.country_dz;
-      case fr:
-        return context.strings.country_fr;
-      case nl:
-        return context.strings.country_nl;
-      case lu:
-        return context.strings.country_lu;
-      case ru:
-        return context.strings.country_ru;
-      default:
-        Log.w('no translation found for country code $this in country.dart');
-        return null;
-    }
-  }
+import 'package:flutter/cupertino.dart';
+
+typedef Localize<String> = String? Function(BuildContext context);
+
+@immutable
+class Country {
+  final String iso2Code;
+  final Localize<String> localize;
+  final List<String> languages;
+  final bool showOffProducts;
```

Comment:
I think `Country` shouldn't decide whether to show OFF products or not - it's rather up for the Suggested Products feature to make this decision.
Imagine if we had 10 location-based features, that would add other 9 fields to `Country`.

Reply:
I knew you would say that :-). 
I agree it's not really the country.dart 's responsibility but i do think having it in country makes things simpeler at the moment (we don't have any other location based features)
* else we need another array to keep track of the enabled countries, while the country_table at the moment only exists for this feature. So i think it is unnecessary complexity to create and maintain another array.

Reply:
I know we already discussed it, and we have opposite opinions on the topic, but I really think we should have a separate array for the list of countries where a features is enabled.
Current approach goes against a lot of software developing principles:
- it goes against the `S` in `SOLID` - responsibility of `Country` is to represent a country, deciding whether a feature should be enabled or not is another responsibility,
- I'm only superficially familiar with the Clean Architecture recommendations, but I know they state a developer shouldn't make entities from the business model to depend on the features a software has,
- Domain-Drive Design principles state a similar thing in this regard.

It's strange that a `Country` depends on a part of the Suggested Products feature.
If for some god-knows-what reason we decide to remove/redo the feature in the future, the `Country` class shouldn't be affected, but it will.

Just imagine an enum `Color`, where the `Color` objects would have a field `showOnMapMarkers`.
If a person would open the project for the first time and then would take a look at the `Color` enum, they would get confused. Same way they would get confused if they would take a look at `Country`: "Why a class from the `lib/model` package depends on 'off products'? What even are the off products? Disabled products?".

If all of the above is not convincing enough, can I then ask you to just trust me? :)
If we add the field now, it's 100% we will have to remove it in the future, so why add it in the first place.

Reply:
If you would like that, we can leave the field here to see how everything will work out in the future :)
My bet is that in a couple of months the field would have to be moved out, but I of course cannot foresee the future.
If we add an according comment like:
```
/// This field serves the Suggested Products feature, some would say it shouldn't be here, but we decided to leave it be to see what happens to it in the future
```
...that would be fine with me. That would be an interesting experiment.

Reply:
I do understand what you are saying from a design pattern point of view and I don't oppose. My point being is we have a country_table now, which only exists to serve the feature. If new features come that require countries, it's quite possible the concept has to be rethought. 
As i see the code now, i see 2 arrays/maps that basically do the same thing, adding a country means maintaining it in 2 places. But i agree that from design point of view it shouldn't be there. 
So i have created the array back in the offshopmanager :-)